### PR TITLE
Fix #10439: [GS] Validate story page button colour, flags, cursor and vehicle type

### LINKED ISSUE #10439: [Crash]: Passing invalid colour, cursor or vehtype
### Version of OpenTTD

13.0-RC2, Windows 11 Pro

### Steps to reproduce

Create a GS with this in the Start function, then start a new game with it:
```
local colour = -1;  // invalid, will crash
local flags = -1;   // invalid, won't crash
local cursor = -1;  // invalid, will crash
local vehtype = -1; // invalid, will crash

/* Crashes OpenTTD */
local reference1 = GSStoryPage.MakePushButtonReference(colour, flags);
local reference2 = GSStoryPage.MakeTileButtonReference(colour, flags, cursor);
local reference3 = GSStoryPage.MakeVehicleButtonReference(colour, flags, cursor, vehtype);
```

### Upload crash files

[crash20230201151546.zip](https://github.com/OpenTTD/OpenTTD/files/10558484/crash20230201151546.zip)

### PR DESCRIPTION
**Pull request by SamuXarick**
## Motivation / Problem
Fix #10439

## Description
Adds `GSStoryPage::IsValidStoryPageButtonColour`, `GSStoryPage::IsValidStoryPageButtonFlags` and `GSStoryPage::IsValidStoryPageButtonCursor` to the API.

Add missing enforced preconditions to validate parameters passed to `MakePushButtonReference`, `MakeTileButtonReference` and `MakeVehicleButtonReference`

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by glx22** [comment]
The `!xxx.ValidateXXX()` are not enough?

**Comment by SamuXarick** [comment]
> The `!xxx.ValidateXXX()` are not enough?

No. There are asserts before that, in data.SetColour(), data.SetCursor() and data.SetVehicleType().

File: src/script/api/script_story_page.cpp [hunk 1]
```diff
@@ -218,10 +218,57 @@ static inline bool StoryPageElementTypeRequiresText(StoryPageElementType type)
...
+/* static */ bool ScriptStoryPage::IsValidStoryPageButtonFlags(StoryPageButtonFlags flags)
+{
+	/* Don't allow float left and right together */
+	if ((flags & SPBF_FLOAT_LEFT) && (flags & SPBF_FLOAT_RIGHT)) return false;
+	/* Don't allow undefined flags */
+	if (flags & ~(SPBF_FLOAT_LEFT | SPBF_FLOAT_RIGHT)) return false;
+	return true;
+}
+
+/* static */ bool ScriptStoryPage::IsValidStoryPageButtonCursor(StoryPageButtonCursor cursor)
```
Comment (glx22) [inline src/script/api/script_story_page.cpp:241]: There's already a `IsValidStoryPageButtonCursor()` in the story_base.h

File: src/script/api/script_story_page.cpp [hunk 2]
```diff
@@ -218,10 +218,57 @@ static inline bool StoryPageElementTypeRequiresText(StoryPageElementType type)
 	return ScriptObject::Command<CMD_REMOVE_STORY_PAGE_ELEMENT>::Do(story_page_element_id);
 }
 
+/* static */ bool ScriptStoryPage::IsValidStoryPageButtonColour(StoryPageButtonColour colour)
```
Comment (glx22) [inline src/script/api/script_story_page.cpp:227]: story.cpp uses `IsValidColours()`

Comment (SamuXarick) [reply src/script/api/script_story_page.cpp:227]: Those are using non-API enumerated constants, I thought I shouldn't be using them.

**Review by glx22 — APPROVED** [review]

**Pull request merged**